org.eclipse.dawnsci.remotedataset.client/src/org/eclipse/dawnsci/remotedataset/client/RemoteLoader.java

```diff
@@ -50,4 +58,40 @@ public IDataset getDataset(IMonitor unused, SliceND slice) throws IOException {
 		return ret;
 	}
 
+	@Override
+	public int[] refreshShape() {
+
+		try {
+			RemoteDatasetInfo datasetInfo = getDatasetInfo();
+			return datasetInfo.getShape();
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			return null;
+		}
+	}
+	
+	public RemoteDatasetInfo getDatasetInfo() throws Exception {
+		List<String> info = getInfo();
+		
+		 return new RemoteDatasetInfo(info);
+		
+	}
+	
+	private List<String> getInfo() throws Exception {
+		
+		final List<String> ret = new ArrayList<String>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)